custom/dfci_swasth/rules/custom_actions.py

```diff
@@ -0,0 +1,81 @@
+from corehq.apps.data_interfaces.models import CaseRuleActionResult
+from corehq.apps.hqcase.utils import update_case, AUTO_UPDATE_XMLNS
+from corehq.form_processor.exceptions import CaseNotFound
+from corehq.form_processor.models import CommCareCase
+from custom.dfci_swasth.constants import (
+    CASE_TYPE_PATIENT,
+    PROP_CCUSER_CASELOAD_CASE_ID,
+    PROP_COUNSELLOR_CLOSED_CASE_LOAD,
+    PROP_COUNSELLOR_LOAD,
+)
+
+
+def update_counsellor_load(patient_case, rule):
+    num_related_updates = 0
+
+    if patient_case.type != CASE_TYPE_PATIENT:
+        return CaseRuleActionResult(num_related_updates=num_related_updates)
+
+    ccuser_caseload_case = _get_ccuser_caseload_case(patient_case)
+
+    # Updating the case type 'ccuser_caseload' case properties
+    if ccuser_caseload_case:
+        case_updates = _get_case_updates(ccuser_caseload_case)
+        if case_updates:
+            (submission, cases) = update_case(
+                ccuser_caseload_case.domain,
+                ccuser_caseload_case.case_id,
+                case_properties=case_updates,
+                xmlns=AUTO_UPDATE_XMLNS,
+                device_id=__name__ + ".update_counsellor_load",
+                form_name=rule.name,
+            )
+            num_related_updates = 1
+            rule.log_submission(submission.form_id)
+
+    return CaseRuleActionResult(
+        num_related_updates=num_related_updates,
+    )
+
+
+def _get_ccuser_caseload_case(patient_case):
+    coun_ccuser_caseload_case_id = patient_case.get_case_property(PROP_CCUSER_CASELOAD_CASE_ID)
+    if coun_ccuser_caseload_case_id:
+        try:
+            return CommCareCase.objects.get_case(coun_ccuser_caseload_case_id, domain=patient_case.domain)
+        except CaseNotFound:
+            return None
+
+
+def _get_case_updates(ccuser_caseload_case):
+    counsellor_load = _get_updated_counsellor_load(ccuser_caseload_case)
+    counsellor_closed_case_load = _get_updated_counsellor_closed_case_load(ccuser_caseload_case)
+
+    result = {}
+    if counsellor_load is not None:
+        result.update({PROP_COUNSELLOR_LOAD: counsellor_load})
+    if counsellor_closed_case_load is not None:
+        result.update({PROP_COUNSELLOR_CLOSED_CASE_LOAD: counsellor_closed_case_load})
+    return result
+
+
+def _get_updated_counsellor_load(ccuser_caseload_case):
+    counsellor_load = _get_integer_case_property_value(ccuser_caseload_case, PROP_COUNSELLOR_LOAD)
+    # Value of counsellor load can be minimum 0
+    if counsellor_load is not None and counsellor_load >= 1:
+        return counsellor_load - 1
+
+
+def _get_updated_counsellor_closed_case_load(ccuser_caseload_case):
+    counsellor_closed_case_load = _get_integer_case_property_value(
+        ccuser_caseload_case, PROP_COUNSELLOR_CLOSED_CASE_LOAD)
+    # Value of counsellor closed case load can be minimum 0
+    if counsellor_closed_case_load is not None and counsellor_closed_case_load >= 0:
```

Comment:
```suggestion
    if counsellor_closed_case_load and counsellor_closed_case_load >= 0:
```

Reply:
Same as https://github.com/dimagi/commcare-hq/pull/33364#discussion_r1314645669

Reply:
👍  Got it.

Reply:
It's strange though that for counsellor_load the check is not for None, I would assume that the reasoning would apply to both?

Reply:
For `counsellor_load`, there is check for it to be one or more, so if its zero, whole condition will still be false . While in case of `counsellor_closed_case_load` it can be zero or more, hence the `None` check as value zero will otherwise evaluate to false.

Reply:
the comment for both is "can be minimum 0", so that needs to change for one of them?

Reply:
Updated comment
https://github.com/dimagi/commcare-hq/compare/2812e5ac55d27a3b34da11e3fdff6122083603dc..48ecc72c9e560e9c7f2e02192f534f2367f31c08